statsmodels/base/distributed_estimation.py

```diff
@@ -0,0 +1,431 @@
+from statsmodels.base.elastic_net import fit_elasticnet
+from statsmodels.base.model import Results
+from statsmodels.tools.decorators import cache_readonly
+from statsmodels.regression.linear_model import OLS
+#from statsmodels.genmod.generalized_linear_model import GLM
+#from statsmodels.genmod.families import Binomial
+import statsmodels.base.wrapper as wrap
+import numpy as np
+
+"""
+distributed regularized estimation.
+
+Routines for fitting regression models using a distributed
+approach outlined in
+
+Jason D. Lee, Qiang Liu, Yuekai Sun and Jonathan E. Taylor.
+"Communication-Efficient Sparse Regression: A One-Shot Approach."
+arXiv:1503.04337. 2015. http://arxiv.org/abs/1503.04337.
+
+Currently the primary usecase is for out of memory computation,
+this approach currently does not support mutli-core processing
+and processing must be handled sequentially.
+
+
+There are several variables that are taken from the source paper
+for which the interpretation may not be directly clear from the
+code, these are mostly used to help form the estimate of the
+approximate inverse covariance matrix as part of the
```

Comment:
In the OLS case it is an approximation of (1 / n X^T X)^{-1}, I've added a note of this in the code.  It is used as part of the debiasing procedure:

beta_tilde = 1 / K sum_k  beta_hat_k + theta_hat 1 / K sum_k nabla l_k(beta_hat_k) / N_k